pandas/tseries/tests/test_timeseries.py

```diff
@@ -3291,6 +3291,12 @@ def test_NaT_scalar(self):
         series[2] = val
         self.assertTrue(com.isnull(series[2]))
 
+    def test_NaT_cast(self):
+        # GH10747
+        result = Series([iNaT]).astype('datetime64[ms]')
```

Comment:
because ```pandas.Series([numpy.nan]).astype('datetime64[ns]')``` and ```pandas.Series(['NaT]).astype('datetime64[ms]')```work?
Now I'm not so sure, but say you have an array of integer ns values, and with some missing data, you can `astype` to `datetime64[ns]` without surprises, but if you have ms values and missing data, then you can't.